resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/PathParamInjector.java

```diff
@@ -109,30 +109,21 @@ else if (pathSegmentList)
             return segments[segments.length - 1];
          }
       }
-      else
+      List<String> list = request.getUri().getPathParameters(!encode).get(paramName);
+      if (list == null)
       {
-         List<String> list = request.getUri().getPathParameters(!encode).get(paramName);
-         if (list == null)
-         {
-            if (extractor.defaultValue == null) throw new InternalServerErrorException(Messages.MESSAGES.unknownPathParam(paramName, request.getUri().getPath()));
-            if (extractor.isCollectionOrArray())
-            {
-               return extractor.extractValues(null);
-            }
-            else
-            {
-               return extractor.extractValue(null);
-            }
-         }
-         if (extractor.isCollectionOrArray())
-         {
-            return extractor.extractValues(list);
-         }
-         else
-         {
-            return extractor.extractValue(list.get(list.size() - 1));
-         }
-      }
+	    if (extractor.defaultValue == null) throw new InternalServerErrorException(Messages.MESSAGES.unknownPathParam(paramName, request.getUri().getPath()));
+	    if (extractor.isInjectableAsCollectionOrArray())
+	    {
+	       return extractor.extractValues(null);
+	    }
+		return extractor.extractValue(null);
+	 }
+	 if (extractor.isInjectableAsCollectionOrArray())
+	 {
+	    return extractor.extractValues(list);
+	 }
+	 return extractor.extractValue(list.get(list.size() - 1));
    }
 
```

Comment:
Ok I get it. It's actually better to use the 'else' block to keep the control on flow. Thank you 👍